lib/spack/docs/packaging_guide.rst

```diff
@@ -3453,6 +3441,13 @@ directory, including cached virtual indices.
 
 To remove all of the above, the command can be called with ``--all``.
 
+When called with positional arguments, cleans up temporary files only
+for a particular package, by deleting the
+expanded/checked out source code *and* any downloaded archive.  If
```

Comment:
I would just remove the second half of the sentence. We've already explained what `spack clean`/`spack clean --staged` does. Here we are clarifying that `spack clean foo` only applies to `foo`, not to every package.